junitparser/junitparser.py

```diff
@@ -344,35 +344,37 @@ def is_passed(self):
         """Whether this testcase was a success (i.e. if it isn't skipped, failed, or errored)."""
         return not self.result
 
+    @property
+    def is_failure(self):
+        """Whether this testcase failed."""
+        return any(isinstance(r, Failure) for r in self.result)
+
+    @property
+    def is_error(self):
+        """Whether this testcase errored."""
+        return any(isinstance(r, Error) for r in self.result)
+
     @property
     def is_skipped(self):
         """Whether this testcase was skipped."""
-        for r in self.result:
-            if isinstance(r, Skipped):
-                return True
-        return False
+        return any(isinstance(r, Skipped) for r in self.result)
 
     @property
     def result(self):
         """A list of :class:`Failure`, :class:`Skipped`, or :class:`Error` objects."""
-        results = []
-        for entry in self:
-            if isinstance(entry, tuple(POSSIBLE_RESULTS)):
-                results.append(entry)
-
-        return results
+        return [r for r in self if isinstance(r, tuple(POSSIBLE_RESULTS))]
 
     @result.setter
     def result(self, value: Union[Result, List[Result]]):
         # First remove all existing results
         for entry in self.result:
-            if any(isinstance(entry, r) for r in POSSIBLE_RESULTS):
+            if isinstance(entry, tuple(POSSIBLE_RESULTS)):
                 self.remove(entry)
         if isinstance(value, Result):
             self.append(value)
         elif isinstance(value, list):
             for entry in value:
-                if any(isinstance(entry, r) for r in POSSIBLE_RESULTS):
+                if isinstance(entry, tuple(POSSIBLE_RESULTS)):
                     self.append(entry)
```

Comment:
I don't get some of the guard in this code?

if passing in a single value it just needs to inherit from `Result` but passing a list will ignore all values that are not `Failure | Error | Skipped`? And why are values potentially not cleared from the previous content of `self`?

Reply:
Right, there is a bit of inconsistency here. The method signature says any `Result` and list of that type can be added, but only those specific sub-types in `POSSIBLE_RESULTS` are actually added, if passed as a list.

Also, the `result` getter filters for `POSSIBLE_RESULTS` as well does the setter when removing them. If @weiwei does not know a particular reason / use case for that, I think this could be simplified to `Result` here as users might want to add custom `Result` elements.

Reply:
I think the results returned here should be distinguished from XUnit2 rerun results: https://github.com/Cube707/junitparser/pull/1